### DISCUSSION TITLE
Initialize livewire and alpinejs manualy

### DISCUSSION THREAD

**Discussion by Briareos17** [livewire/livewire#4433] [1. Help]
I'm trying to lazy load livewire and alpine libs after page completely loads. So, I have some progress in here, but, one issue, that alpine do not see livewire's methods, like emitTo.
Here is my loader:

```
<script defer {{ $attributes }}>
    function loadLivewire() {
        let script = document.createElement('script');
        script.src = '{{ $fullAssetPath }}';
        script.setAttribute('data-turbo-eval', 'false');
        script.setAttribute('data-turbolinks-eval', 'false');
        document.body.append(script);

        setTimeout(() => {
            window.Livewire = new Livewire();
            window.Livewire.devTools({{ var_export($shouldDebug) }});
            window.livewire_app_url = '{{ $appUrl }}';
            window.livewire_token = '{{ $jsLivewireToken }}';
            window.Livewire.start();

            window.deferLoadingAlpine = function (callback) {
                window.addEventListener('livewire:load', function () {
                    callback();
                });
            };

            window.addEventListener("alpine:initializing", function () {
                window.Livewire.start();
            });

            loadAlpine();
        }, 50);
    }

    function loadAlpine() {
        document.addEventListener('alpine:init', () => {
            Alpine.store('app', {
                sidebarOpen: false,
                selectedRepresentative: null,
                formOpen: null,
                modalOpen: null,
                recaptchaSitekey: null,
                addressesModalOpen: null,
                recallOpen: null
            });
        });

        let script = document.createElement('script');
        script.src = 'https://unpkg.com/alpinejs@3.4.1/dist/cdn.min.js';
        document.body.append(script);
    }

    window.addEventListener('load', () => {
        setTimeout(() => {
            loadLivewire();
        }, {{ $timeout }});
    });
</script>
```

All fine unless i click on some button that uses livewire logic, and then I got this:

```
VM128705:3 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'emitTo')
    at eval (eval at generateFunctionFromString (cdn.js:380), <anonymous>:3:48)
    at cdn.js:391
    at tryCatch (cdn.js:415)
    at cdn.js:2595
    at handler3 (cdn.js:2054)
    at cdn.js:2075
    at cdn.js:2056
    at cdn.js:2097
    at HTMLButtonElement.<anonymous> (cdn.js:2056)
```

Is there are any method to manualy make alpine to recognize livewire integration?

**Comment by joshhanley** [comment]
@Briareos17 hmm that's not really supported. Why are you trying to lazy load Livewire and Alpine?

**Comment by Briareos17** [comment]
Because our website has terrible GTM with ton of tags and make huge impact to page load perfomance, so I trying to lazy load all available scripts because I can't lazy load GTM script.
I just can't figure out - what the difference between default livewire algorithm and lazy load with same order? May be I need to handle some event in order to livewire recognize alpine loading and inject it's directives into alpine?

Literally I almost make it success, alpine and livewire works perfect separately, I just need some mechanism to inject livewire directives into alpine.

**Comment by Briareos17** [comment]
Any suggestions?

**Reply by joshhanley** [reply]
@Briareos17 sorry I've got no suggestions 😕 Livewire's Alpine support should be built in to Livewire's compile script file, so when Livewire's script runs, it should inject what it needs into Alpine, so as long as Alpine is on that page but hasn't started it should work.

**Reply by Briareos17** [reply]
So,I need to add Alpine on page efore livewire but start after? I will try, thanks.

**Reply by joshhanley** [reply]
@Briareos17 yeah maybe, I'm a bit rusty on the operation order
